src/WebJobs.Extensions.DurableTask/DurableActivityContext.cs

```diff
@@ -102,6 +103,31 @@ internal object GetInput(Type destinationType)
             }
 
             string serializedValue = jToken.ToString(Formatting.None);
+
+            // Object inputs for out-of-proc activities are passed in their JSON-stringified form
+            // with a destination type of System.String. Unfortunately, deserializing a JSON string
+            // to a string causes MessagePayloadDataConverter to throw an exception. This is a
+            // workaround for that case.
+            if (destinationType.Equals(typeof(string)))
+            {
+                // Json.NET has no TryParse or IsValidJson method, so we'll optimistically parse
+                // and catch any exceptions (which shouldn't occur: all (in-proc; JSON and not JSON)
+                // or all other (out-of-proc: not-JSON) inputs with destination System.String should
+                // cast to valid JValues and be handled above.)
+                try
+                {
+                    var deserializedValue = MessagePayloadDataConverter.Default.Deserialize(serializedValue, typeof(object));
+                    if (deserializedValue != null)
+                    {
+                        return serializedValue;
+                    }
+                }
+                catch (JsonReaderException ex)
+                {
+                    throw ex;

```

Comment:
There is no need for a `catch` block which simply rethrows the exception. In fact, there is a problem with this as-written because the original call stack information will be lost when you rethrow the exception this way (normally one is advised to use `throw` without any arguments to rethrow an exception). But anyways, I would remove the try/catch entirely.